src/hyperlink/_url.py

```diff
@@ -1122,7 +1194,7 @@ def from_text(cls, text):
         port = au_gs['port']
         if port is not None:
             try:
-                port = int(port)
+                port = int(port)  # type: ignore[assignment] FIXME, also below
```

Comment:
When we set port a few lines up, it's a string, and now we're setting it to an `int`.
That should be cleaned up, as currently port is sometimes a string when it is passed to `cls()`, and I think that's incorrect.
I made some attempts are fixing it, and kept running into problems, so it probably should go in a separate PR.